src/gui/codeeditors/qgscodeeditorpython.cpp

```diff
@@ -188,14 +203,153 @@ void QgsCodeEditorPython::initializeLexer()
 
 void QgsCodeEditorPython::keyPressEvent( QKeyEvent *event )
 {
+  // If editor is readOnly, use the default implementation
+  if ( isReadOnly() )
+  {
+    return QgsCodeEditor::keyPressEvent( event );
+  }
   const bool ctrlModifier = event->modifiers() & Qt::ControlModifier;
 
+  // Toggle comment when user presses  Ctrl+:
   if ( ctrlModifier && event->key() == Qt::Key_Colon )
   {
     event->accept();
     toggleComment();
     return;
   }
+
+  const QgsSettings settings;
+
+  bool autoCloseBracket = settings.value( QStringLiteral( "/pythonConsole/autoCloseBracket" ), true ).toBool();
+  bool autoInsertImport = settings.value( QStringLiteral( "/pythonConsole/autoInsertImport" ), false ).toBool();
+
+  // Update calltips when cursor position changes with left and right keys
+  if ( event->key() == Qt::Key_Left  ||
+       event->key() == Qt::Key_Right ||
+       event->key() == Qt::Key_Up ||
+       event->key() == Qt::Key_Down )
+  {
+    QgsCodeEditor::keyPressEvent( event );
+    callTip();
+    return;
+  }
+
+  // Get entered text and cursor position
+  const QString eText = event->text();
+  int line, column;
+  getCursorPosition( &line, &column );
+
+  // If some text is selected and user presses an opening character
+  // surround the selection with the opening-closing pair
+  if ( hasSelectedText() )
```

Comment:
```suggestion
  if ( hasSelectedText() && autoCloseBracket )
```

Reply:
If you want to be able to disable the quote wrapping feature, what do you think about having two separate settings: `autoSurround `and `autoCloseBracket`? I think that's how vscode does it.

Reply:
Sounds good to me, thanks!